cql3/restrictions/statement_restrictions.cc

```diff
@@ -1340,5 +1340,59 @@ const single_column_restrictions::restrictions_map& statement_restrictions::get_
     return single_restrictions->restrictions();
 }
 
+std::vector<query::clustering_range> statement_restrictions::get_global_index_clustering_ranges(
+        const query_options& options,
+        const schema& idx_tbl_schema,
+        const column_definition& indexed_column) const {
+    if (!_partition_range_is_simple) {
+        on_internal_error(rlogger, "get_global_index_clustering_ranges: complex partition key");
+    }
+    // This assumes that a base-table partition column cannot be the indexed column:
+    std::vector<expression> idx_tbl_ck_prefix(1 + _schema->partition_key_size());
+    idx_tbl_ck_prefix.reserve(idx_tbl_schema.clustering_key_size());
```

Comment:
Up to here should be done in prepare phase.

Reply:
Done.